### ISSUE TITLE
[Generic-Pass]: Need to set the  background image properties

### ISSUE THREAD

**Issue by prashant03Dev** [alexandercerutti/passkit-generator#195]
## Running OS

Windows,Mac

## Running Node Version

Version : v18.20.0
## Description

-As we have created a generic pass and loaded all the images as per the size provided in the package.But currently We need to change some background properties.

## Expected behavior
	Background properties should be change as per the user request

Source Code:

 const passCreated: PassKit.PKPass = await this.walletService.createPass(
            'Generic.pass',
            this.user.uuid,
            'membership-card-pass',
        );

        try {
            passCreated.type = this.pass.type;

            passCreated.props.beacons
            passCreated.primaryFields.push({
                key: 'memberName',
                label: 'Member Name',
                value: `${this.user.insuredName}`,
                textAlignment: 'PKTextAlignmentRight',
            });

            passCreated.secondaryFields.push({
                key: 'memberNumber',
                label: 'Member Number',
                value: `${this.user.memberId}`,
                textAlignment: 'PKTextAlignmentRight',
            });

@alexandercerutti Suggest us the way to set the background properties dynamically or in pass.json file.

**Comment by alexandercerutti** [comment]
Hi @prashant03Dev, thanks for using passkit-generator!

In the API documentation (which I can suppose you didn't read or didn't see), there is a beautiful 🌟 method called `addBuffer()`, that I think might suit you needs 😉

Let me know.

**Comment by prashant03Dev** [comment]
Hi Alex,

Thanks for mentioning method for better clarification,But as I can see
there are not any methods to set the background related changes,

As of now the generic pass image that i stored,is getting blurriness,So to
remove that I'm getting troubling,

Can you suggest me the method or a way to do that!

Thanks

**Comment by alexandercerutti** [comment]
Sorry but I don't have clear your exact use case.

You can set a `backgroundImage` file but you cannot remove the blur effect. That's an Apple feature for passes that support the background image (not all of them does). This is what [Documentation Archive says](https://developer.apple.com/library/archive/documentation/UserExperience/Conceptual/PassKit_PG/Creating.html#//apple_ref/doc/uid/TP40012195-CH4-SW1) about:

> The background image (background.png) is displayed behind the entire front of the pass. The expected dimensions are 180 x 220 points. The image is cropped slightly on all sides and blurred. Depending on the image, you can often provide an image at a smaller size and let it be scaled up, because the blur effect hides details. This lets you reduce the file size without a noticeable difference in the pass.

Is "removing the blur" the only thing you were trying to achieve?

**Comment by prashant03Dev** [comment]
Hi Alex,

Is there any pass which has the background without blur effect?

**Comment by alexandercerutti** [comment]
I don't think so. I think the idea of Apple is that a blurred image helps to improve readability.

**Comment by alexandercerutti** [comment]
So, can we close the issue? :)

**Comment by prashant03Dev** [comment]
Yes, please!

**Comment by alexandercerutti** [comment]
Glad I could help you! If you could leave a 🌟 on the project, that would be awesome! 🚀

**Comment by prashant03Dev** [comment]
> Glad I could help you! If you could leave a 🌟 on the project, that would be awesome! 🚀

🌟
